app/workers/emotion_selection_notification_worker.rb

```diff
@@ -9,13 +9,15 @@ def initialize
   def run_notification
     return unless Date.current.strftime('%A').casecmp?(ENV.fetch('DAY_TO_SEND_INVITES'))
 
+    @time_period.due_date = Date.current
+    @time_period.save
     run_notification!
   end
 
   private
 
   def run_notification!
-    @users.each { |user| UserEmailMailer.response_invite(user, time_period).deliver_now }
+    @users.each { |user| UserEmailMailer.response_invite(user, @time_period).deliver_now }
```

Comment:
@VitaliiShevchenko you can use `time_period` without `@`, because `time_period `defined in `attr_reader :users, :time_period`

Reply:
I think I can't to use without @ . Will be wrong works because you created such methods:
  def initialize
    @users = User.opt_in
    find_or_create_time_period
  end

  def find_or_create_time_period
    @time_period = TimePeriod.current || TimePeriod.create_time_period
  en

Reply:
Thanks, I check it and  all work, you've right). I changed with your tooltip